### DISCUSSION TITLE
Frage zur Identifikation

### DISCUSSION THREAD

**Discussion by Rustimation** [philippoo66/optolink-splitter#38] [Q&A] [answered]
Hallo, kann ich hier Fragen stellen?
Mit Hilfe der [Anleitung bei ViessData21](https://github.com/philippoo66/ViessData21/blob/master/README.md#datapoints_readme) habe ich meine Gerätekennung ausgelesen: 20cb28cb0000010a
Es handelt sich um eine Vitodens 200-W B2HB 1,9-19kW Baujahr 2019. Steuerung ist eine Vitotronic 200.
Aber irgendwie hänge ich fest. 20CB ergibt laut DataPoints.txt bzw. Devices.csv ca.16 verschiedene Geräte. Der SW Index Ident ist "cb" ist aber in DataPoints.txt nirgends zu finden. 
Kann mir wer auf die Sprünge helfen?
Danke und viele Grüße
Chris

**Comment by philippoo66** [comment]
Moin Chris!

20cb ist klar. Dann kommt das 'eins für alles' (01), d.h. die 28 kannst du ignorieren. dann kommt cb, das liegt zwischen c8 und ff, das heisst du brauchst noch das ecnsysDeviceIdentF0 an Adresse F0 (1 Byte). Damit ergibt sich dann, welche von den  VScotHO1_200_... zu deiner Therme gehört. Das hast du ja noch garnicht ausgelesen mit den 8 Bytes ab F8 ;-)

Ich tippe auf die VScotHO1_200_11, weil die Vitodens 200 hat eine HO1B, und die _11 ist für Geräte ab 2016...

Grüsse!
Phil

**Reply by Rustimation** [reply]
babysteps forward... an F0 steht "0b". Sollte die ecnsysDeviceIdent in der Tabelle hexadezimal sein, stehe ich auf dem Schlauch, weil keins der Intervalle passt. Ist die Tabelle dezimal, dann wäre es wohl Zeile 168: ecnsysDeviceIdentF0:11-19 : VScotHO1_200_11 weil "0b" dezimal 11 entspricht - also analog deines obigen Verdachts. Sehr verwirrend das.

**Reply by philippoo66** [reply]
0b ist tatsächlich nicht in der Tabelle (die Werte sind alle hex), das ist wahrlich seltsam, dass es bei dir an F0 steht. bist du sicher, dass du es nicht vorher versehentlich umgeschrieben hast (das geht nämlich)?

aber mach dir nicht zu viele Gedanken - die Unterschiede_10 und _11 sind minimal was Adressen anbetrifft (auf den ersten Blick seh ich nur hinzugekommene Einmalladung 0x65F5). Die Listen sind eh nicht vollständig (s. das Readme dazu).

Probiert doch mal, ob es die 65F5 bei dir gibt

**Reply by Rustimation** [reply]
Umgeschrieben habe ich bestimmt nichts. War ja bisher ganz braver Viessmann API Nutzer.
65F5:  ja gibt es. Ergebnis ist "00" oder heißt das, es gibt kein Ergebnis? Einmalladung kann meine Vitodens.

**Reply by philippoo66** [reply, marked as answer]
wenn du 65F5 auf 1 setzt (ich meine es ist 1) startest du die WW Einmalladung. 

Wenn du versuchst was zu lesen, was es nicht gibt, kommt eine Fehlermeldung (3 vorne in der Antwort, Code in den Daten).

**Reply by Rustimation** [reply]
okay. Das hilft. Dann habe ich wohl die VScotHO1_200_11
Danke nochmal.

**Reply by philippoo66** [reply]
noch schnell die Return Codes (sind die eigentlich irgendwo im Wiki aufgeführt?):

ReturnCode: 01=success, 03=ErrMsg, 15=NACK, 20=UnknB0_Err, 41=STX_Err, AA=HandleLost, FD=PlLen_Err, FE=CRC_Err, FF=TimeOut (all hex)

vielleicht gibt es inzwischen noch mehr, aber haupt sächlich interessiert ob 01 oder was andres....

**Reply by Rustimation** [reply]
Schnelle Volltextsuche hat ergeben: Steht nicht im Wiki

**Reply by philippoo66** [reply]
doch, grad gefunden ;-)
https://github.com/philippoo66/optolink-splitter/wiki/010-Command-Syntax#response-to-requests

**Reply by Rustimation** [reply]
Hi Phil,
ich schreibe gerade etwas über den Optolink-Splitter in meinem Blog. Da sich Manches in deiner Github Doku nicht so ohne weiteres erschließt, will ich ein paar Punkte etwas ausführlicher behandeln, damit auch wohlmeinende Amateure damit umgehen können: Jetzt geht es um die Identifikation des Geräts, die auch mir noch nicht 100% klar ist:

Mein Gerät hat bei 0xF8 folgende Info: 20cb28cb0000010a
Du schreibst weiter oben:
_20cb ist klar. Dann kommt das 'eins für alles' (01), d.h. die 28 kannst du ignorieren. dann kommt cb, das liegt zwischen c8 und ff, das heisst du brauchst noch das ecnsysDeviceIdentF0 an Adresse F0 (1 Byte). Damit ergibt sich dann, welche von den VScotHO1_200_... zu deiner Therme gehört. Das hast du ja noch garnicht ausgelesen mit den 8 Bytes ab F8 ;-)_

**Frage**: Woher kommt das 'eins für alles'? An welcher Adresse steht das?

Hast du außer deiner Identifikation noch vielleicht andere Beispiele, anhand derer man das Identifizieren verdeutlichen kann?

Danke für deine Mühen. Ich würde mich freuen, wenn du meinen Beitrag ggf. querchecken könntest.

VG
Chris

**Reply by philippoo66** [reply]
hai Chris!

Wenn du was Nützliches hast, kannst du das auch gern in's Wiki schreiben, das ist offen. Das mit der deviceIdent gehört wohl eher etwas nach hinten, weil es ja nur indirekt mit dem Splitter zu tun hat. Wobei Sinn machen würde es ja in Zusammenhang mit Frans' Datapoint Adressen (310).

 > Woher kommt das 'eins für alles'?

das hab ich mir so ausgedacht - als Eselsbrücke ;-)) Es steht nirgendswo (jedenfalls nicht, dass ich wüsste), es ist halt einfach so, zumindest bei den idents, die mir bisher untergekommen sind. es gab noch keins, wo an der Stelle 01 stand. Leider hab ich da bisher keine Notizen zu gemacht, das müsste man aus diversen Threads zusammensammeln.

poste gerne mal den Link auf deinen Blog, dann kann ich da ja gucken (und auch den Rest vom Blog - interessiert mich immer sowas). Ggf. kann ich auch vorher drüber gucken, wenn dir das lieber ist. Eine 'öffentliche' e Adr ist open3e ad web/punkt/de wegen dem anderen Viessmann Projekt, das aufgrund der aktuellen Generation schon etwas mehr Publizität geniesst.

**Reply by Rustimation** [reply]
Danke für deine Antwort. War die letzten Tage unterwegs, deshalb erst heute eine Antwort.
Heißt das, dass der Hardwareindex bei 0xFA generell ignoriert werden kann? Das wäre also das "1 für all"e, egal was da steht... Richtig?
Ich finde die deviceIdent ist schon wichtig, sobald man die spezielleren Datenpunkte für seine eigene Heizung braucht und in der passenden Tabelle nachsehen will. Der Überlappungsgrad zwischen den einzelnen DP Listen ist zwar sehr groß aber gerade zwischen Gas, Holz und WP Geräten dürfte sich da doch Einiges unterscheiden. 
Seit heute Vormittag ist die API wieder offline. Ich werde mich jetzt etwas beeilen und mein Dashboard komplett auf den Optolink Splitter umstellen. 
VG aus München
Chris

**Reply by philippoo66** [reply]
hier hat es ein paar Geräte, wo der HardwareIndex wohl nicht egal ist

![grafik](https://github.com/user-attachments/assets/d7274ea8-5c50-448f-8298-72ee628ac37b)

die Geräte sind mir allerdings noch nicht untergekommen, von daher kann ich nicht sagen, ob dann da wirklich was mit 1, 2, 8 oder 10 steht und/oder wie es sich da verhält.

ich kann auch nicht sagen ob meine (aus den mir vorliegenden Vitosoft Daten gewonnene) Devices Tabelle komplett vollständig ist. Soweit ich weiss, bin ich überhaupt der erste und einzige, der eine Aufschlüsselung/Zuordnung der deviceIdent überhaupt mal 'in den Raum gestellt' hat.

Wie gesagt, Viessmann hat (auch) dazu meines Wissens nichts, zumindest nicht öffentlich zugänglich, dokumentiert. Von daher müssen wir mit dem leben, was wir 'empirisch herausfinden'. Einen Anspruch auf Vollständigkeit oder uneingeschränkte Richtigkeit gibt es nicht. Sollten irgendwo ergänzende oder andere Informationen auftauchen, würde ich mich freuen, davon Kenntnis zu erhalten!

Wir könnten versuchen, die Viessdata.exe zu decompilieren und noch weiter was aus den SQL Daten zu gewinnen. Die Tabelle stammt aus den xml Daten soweit ich erinnere. Ich hatte den Eindruck, dass die sich großteils mit den SQL Daten decken, aber was Frans immer so zutage bringt, deutet ja an, dass in der SQL noch mehr ist...

**Reply by jaegerschnitzel** [reply]
> babysteps forward... an F0 steht "0b". Sollte die ecnsysDeviceIdent in der Tabelle hexadezimal sein, stehe ich auf dem Schlauch, weil keins der Intervalle passt. Ist die Tabelle dezimal, dann wäre es wohl Zeile 168: ecnsysDeviceIdentF0:11-19 : VScotHO1_200_11 weil "0b" dezimal 11 entspricht - also analog deines obigen Verdachts. Sehr verwirrend das.

Ich habe auch "0b" unter F0. Seltsam, dass wir die einzigen sind.
Danke für deine Blog-Artikel. Diese waren sehr hilfreich für mich. Allerdings kann ich deine Auswahl der Datenpunkte nicht ganz nachvollziehen ;-) Bin aber noch selber am schauen und poste meine dann bei dir im Blog. Eventuell hast du was übersehen oder dich interessieren gewisse Dinge einfach nicht.

**Reply by Rustimation** [reply]
Schön, wenn dir mein Blog geholfen hat. Meine Vitodens 200-W (B2HB 1,9 - 19kW) wurde in Nordtalien gekauft und und wird auch dort betrieben, vielleicht steht das 0b für eine spezielle Landesversion?
In erster Linie wollte ich mein [Dashboard](https://www.rustimation.eu/wordpress/wp-content/uploads/2023/01/2023-02-15-16_28_33-Centralina-%E2%80%93-Mozilla-Firefox.png), das ich auf Basis der API gebaut hatte komplett mit dem Optolink Splitter versorgen. Viel mehr habe ich seinerzeit nicht herauslesen können bzw. auch nicht benötigt. 
Die Puffertemperatur wird zwar von Viessmann an vier Punkten ausgelesen, die beiden mittleren Sensoren kann man aber nicht abfragen, ebensowenig wie die allgmeine Rücklauftemperatur oder den Vorlauf vom Puffer zur Heizung. Ich habe mir dann mit DS18B20 Steck- bzw. Anlegefühlern geholfen. Einstellungen der Heizkurve und Zeitprogramme mache ich weiterhin über die App, da sich der Aufwand für die wenigen Male nicht lohnt.
Trotzdem wäre ich - und Phils Github Leserschaft - sehr an entschlüsselten Datenpunkten interessiert.
Viele Grüße und ein gutes neues Jahr.
Chris

**Comment by Rustimation** [comment]
oops. war ich mal wieder zu ungenau bzw. die Suchleiste oben rechts gibt das nicht her... "Return Codes" zeigt nicht auf das Wiki

**Comment by lellem** [comment]
Ich mache hier mal weiter mit einer weiteren Frage zur Identifikation. Vielleicht ja jemand eine Erklärung...

Zunächst mal vielen Dank an @philippoo66 für den Optolink-Splitter und die Mühe die du dir damit gemacht hast.

Ich habe eine Vitocal 200-A AWO-E-AC 201.A16 und habe die Identifikation 0x204d01410000050a ausgelesen. Dies ist ja laut Tabelle ein CU401B_S, was ja demzufolge eine Vitocal xxx-S mit Vitotronic 200 (Typ WO1C) sein sollte, was aber wiederum nicht zu meiner Anlage passt.

Hat hierzu jemand eine Erklärung?

**Reply by philippoo66** [reply]
moin @lellem !

danke für die Blumen :-) Zu deiner Frage fällt mir nur in Anlehnung an ein altes Buch ein

_"Die Wege der Viessmänner sind unergründlich."_ ;-)

Hast du denn die passenden bzw dich interessierenden Parameter gefunden?

**Reply by lellem** [reply]
Ich bin noch nicht ganz fertig und eher noch im Testbetrieb. Die meisten Parameter die in der Config hinterlegt waren haben erstaunlicherweise gepasst, was im übrigen auch nicht zu der ausgelesen ID passt.

Aber ich mache dazu demnächst ein neue Diskussion auf.

**Comment by f-brandt** [comment]
Hallo zusammen,
evtl. kann mir auch jemand die passende ID verraten, so ganz sicher bin ich mir nicht und gemäß der Anleitung von Chris hilft wohl nur fragen. Es handelt sich um eine Vitodens 200 HO1B von ca. 2015 (kann auch 2016 sein). Falls es weiter hilft bzw. nötig ist hätte ich auch die Produktnummer.

Nach der Ausgabe von:
cmnd = read;0xF8;8
resp = 1;0xf8;20cb28cc0000010a

müsste es wohl eine dieser sein:
ecnsysDeviceIdent:20CB sysHardware/SoftwareIndexIdent:01C8-01FF ecnsysDeviceIdentF0:0-0 : VScotHO1_200
ecnsysDeviceIdent:20CB sysHardware/SoftwareIndexIdent:01C8-01FF ecnsysDeviceIdentF0:1-9 : VScotHO1_200_01
ecnsysDeviceIdent:20CB sysHardware/SoftwareIndexIdent:01C8-01FF ecnsysDeviceIdentF0:10-10 : VScotHO1_200_10
ecnsysDeviceIdent:20CB sysHardware/SoftwareIndexIdent:01C8-01FF ecnsysDeviceIdentF0:11-19 : VScotHO1_200_11
ecnsysDeviceIdent:20CB sysHardware/SoftwareIndexIdent:01C8-01FF ecnsysDeviceIdentF0:20-29 : VScotHO1_200_20
ecnsysDeviceIdent:20CB sysHardware/SoftwareIndexIdent:01C8-01FF ecnsysDeviceIdentF0:30-39 : VScotHO1_200_30

Nach der Ausgabe von:
cmnd = read;0xF0;8
resp = 1;0xf0;0a00ffffffffff20

Demnach müsste es entweder: 
für das 00 hinter 0a die: 

ecnsysDeviceIdent:20CB sysHardware/SoftwareIndexIdent:01C8-01FF ecnsysDeviceIdentF0:0-0 : VScotHO1_200

oder für die 20 am Ende die 

ecnsysDeviceIdent:20CB sysHardware/SoftwareIndexIdent:01C8-01FF ecnsysDeviceIdentF0:20-29 : VScotHO1_200_20 

sein, habe ich das richtig interpretiert?

**Reply by philippoo66** [reply]
moin!

der F0 ist nur ein einziges Byte, also die 0a hier. 

Da in der Liste nach 9 die 10 kommt und danach die 11, vermute ich dass hier dezimal gezählz wird. 0a ist ja 10 dez, von daher würde ich auf VScotHO1_200_10 tippen.

Hast du denn schon mal die Dateien verglichen, ob es da überhaupt wesentiche Unterschiede gibt?

Grüsse!

**Reply by f-brandt** [reply]
Hi,
ja so groß sind die Unterschiede nicht wirklich. Im Prinzip ist fast alles, von daher fast egal welche Datei man nimmt. 
Vielen Dank.
